theano/compile/function_module.py

```diff
@@ -676,6 +679,17 @@ def __call__(self, *args, **kwargs):
         elif self.unpack_single and len(outputs) == 1:
             return outputs[0]
         else:
+
+            if self.output_dictionary_flag: 
+                outputDict = {}
+
+                assert len(self.output_keys) == len(outputs)
+
+                for i in range(0, len(self.output_keys)): 
+                    outputDict[self.output_keys[i]] = outputs[i]
+
+                outputs = outputDict
```

Comment:
Looks to me like you're re-building the dictionary you had in the first place... Why not just provide this dictionary in ``outputs``? (NB: I  haven't looked closely enough at the code to verify it would work)